api/extensions/ext_redis.py

```diff
@@ -260,7 +260,8 @@ def wrapper(*args, **kwargs):
             try:
                 return func(*args, **kwargs)
             except RedisError as e:
-                logger.warning("Redis operation failed in %s: %s", func.__name__, str(e), exc_info=True)
+                func_name = getattr(func, "__name__", "Unknown")
```

Comment:
```
func_name = func.__name__ or "Unknown"
```
?

Reply:
It's `func.__name__` where violates the rule of unresolved attr. And we have to work around it.

```
error[unresolved-attribute]: Type `(...) -> Unknown` has no attribute `__name__`                                                                                        
   --> extensions/ext_redis.py:263:68
    |
261 |                 return func(*args, **kwargs)
262 |             except RedisError as e:
263 |                 logger.warning("Redis operation failed in %s: %s", func.__name__, str(e), exc_info=True)
    |                                                                    ^^^^^^^^^^^^^
264 |                 return default_return
    |
info: rule `unresolved-attribute` is enabled by default
```